docs/string_grader.md

```diff
@@ -1,66 +1,236 @@
+Note that this will even accept a blank (empty) response. To reject empty responses, you can instead use the `accept_nonempty` flag, which requires at least one character to be submitted (after input cleaning).
+
+```pycon
+>>> grader = StringGrader(
+...     answers={'expect': '', 'grade_decimal': 1, 'msg': 'Your answer has been recorded.'},
+...     accept_nonempty=True,
+...     explain_minimums=None
+... )
+>>> grader(None, 'dog') == {'grade_decimal': 1,
+...                         'msg': 'Your answer has been recorded.',
+...                         'ok': True}
+True
+>>> grader(None, '') == {'grade_decimal': 0, 'msg': '', 'ok': False}
+True
+
+```
 
+Note that when either `accept_any` or `accept_nonempty` are set to True, you do not need to provide any answer to check against.
+
+You may want students to have to write a certain amount of characters or words in order to get credit. Two flags are available to facilitate this: `min_length` and `min_words`, which set a minimum number of characters and words to be awarded credit, respectively (both default to zero). You can use both of these options together if desired.
+
+```pycon
+>>> grader = StringGrader(
+...     answers={'expect': '', 'grade_decimal': 1, 'msg': 'Your answer has been recorded.'},
+...     accept_any=True,
+...     min_length=10,  # Require at least 10 characters (after cleaning input)
+...     min_words=3,    # Require at least 3 words
+...     explain_minimums='msg'
+... )
+>>> grader(None, 'This is a long answer') == {'grade_decimal': 1,
+...                                           'msg': 'Your answer has been recorded.',
+...                                           'ok': True}
+True
+>>> grader(None, 'too short') == {'grade_decimal': 0,
+...                               'msg': 'Your response is too short (2/3 words)',
+...                               'ok': False}
+True
 
-## Accepting Anything
+```
 
-Sometimes you may just want to accept anything that a student provides. There are two options that allow you to do this. The first is `accept_any`, which will literally accept anything that is entered into the textbox. The second is `accept_nonempty`, which requires the textbox to contain a non-empty submission (blank spaces are stripped). `accept_any` has priority over `accept_nonempty`.
+Note that punctuation doesn't break a word for the purpose of word counting, so `isn't word-counting fun?` will only count as three words. If `accept_nonempty` and `min_length` are both used, the more stringent requirement is the one that is used.
+
+When a student's answer is rejected because it doesn't meet the minimum requirements, there are three types of feedback that you can provide, controlled by the `explain_minimums` flag:
+
+* The student receives an error message describing how many words/characters they have, compared to how many are required. This does not consume an attempt (`explain_minimums='err'`, default)
```

Comment:
I just noticed that the `accept_nonempty=True` example above did not raise an error. Should it? Conceptually it falls int he case of "When a student's answer is rejected because it doesn't meet the minimum requirements..."

Maybe `accept_nonempty=True` should internally just modify the configuration during `__init__` so that `StringGrader(accept_nonempty=True)` is exactly equivalent to `StringGrader(accept_any=True, min_length=1)`?

Reply:
I had `explain_minimums=None` turned on, which explains your first statement. As for the second, `accept_nonempty` internally sets `accept_any=True` and `min_length = max(1, min_length)`. I would get rid of `accept_nonempty` completely, were it not for backwards compatibility...

Reply:
Ah, I should hav noticed that `explain_minimums=None`. My bad.

Reply:
Note that throughout the docs, I have explanations set to all of `None`, `err` and `msg`, so each has an example of its behavior. Let me know if you think I should change the above though.